File: src/main/scala/com/fowlart/main/state/Order.scala
package com.fowlart.main.state

import com.fowlart.main.state.cosmos.Item

case class Order (orderId:String,
                  date: String,
                  userId: String,
                  userName: String,
                  userPhoneNumber: String,
                  orderBucket: Set[Item],
                  isDelivered: Boolean) {

  override def toString: String = {
    val bucketString = if (orderBucket.nonEmpty) s"${orderBucket.map(i => s"[${i.name} - ${i.qty}] ").reduce((s1, s2) => s"$s1$s2")}" else s"[]"
    s"""
       |orderId: $orderId
       |date: $date
       |userId: $userId
       |userName: $userName
       |userPhoneNumber: $userPhoneNumber
       |userId: $userId
       |orderBucket: $bucketString
       |""".stripMargin
  }
}

File: src/main/java/com/fowlart/main/state/cosmos/Order.java
package com.fowlart.main.state.cosmos;
import org.springframework.data.annotation.Id;

import java.util.Set;

public class Order  {

    @Id
    private String orderId;
    private String date;
    private String userId;
    private String userName;
    private String userPhoneNumber;
    private Set<Item> orderBucket;

    public Order() {
    }

    public String getOrderId() {
        return orderId;
    }

    public void setOrderId(String orderId) {
        this.orderId = orderId;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getUserPhoneNumber() {
        return userPhoneNumber;
    }

    public void setUserPhoneNumber(String userPhoneNumber) {
        this.userPhoneNumber = userPhoneNumber;
    }

    public Set<Item> getOrderBucket() {
        return orderBucket;
    }

    public void setOrderBucket(Set<Item> orderBucket) {
        this.orderBucket = orderBucket;
    }
}
